etna/analysis/eda_utils.py

```diff
@@ -194,25 +197,69 @@ def sample_acf_plot(
     figsize:
         size of the figure per subplot with one segment in inches
     """
+    nan_flag_segments = False
     if segments is None:
+        nan_flag_segments = True
         segments = sorted(ts.segments)
 
+    plot = plot_pacf if partial else plot_acf
+    title = "Partial Autocorrelation" if partial else "Autocorrelation"
+
     k = min(n_segments, len(segments))
```

Comment:
It seems like it is not really correct. As I understand, we should ignore `n_segments` parameter if `segments` is set. Look at `distribution_plot`.